flows.go

```diff
@@ -158,18 +145,29 @@ func FlowFromEndpoints(src, dst Endpoint) (_ Flow, err error) {
 
 // FastHash provides a quick hashing function for a flow, useful if you'd
 // like to split up flows by modulos or other load-balancing techniques.
-// It uses a variant of Fowler-Noll-Vo hashing, and is guaranteed to collide
-// with its reverse flow.  IE: the flow A->B will have the same hash as the flow
-// B->A.
+// It  is guaranteed to collide with its reverse flow.  IE: the flow A->B
+// will have the same hash as the flow B->A.
+//
+// It creates the hash by first, for each endpoint, xor-ing the bytes of the
+// address in a 32-bit round robin fashion (hash ^= addr[0],hash ^= addr[1]<<8,
+// ...,hash ^= addr[5]<<8). The result for each endpoint is then concatenated
+// together in an order independent way. For addresses less than or equal to
+// 32-bits this guarantees collision free hashes. IE: the flows A->B and B->A
+// will have a hash that no other combination of addresses can produce.
 //
 // The output of FastHash is not guaranteed to remain the same through future
 // code revisions, so should not be used to key values in persistent storage.
-func (f Flow) FastHash() (h uint64) {
-	// This combination must be commutative.  We don't use ^, since that would
-	// give the same hash for all A->A flows.
-	h = fnvHash(f.src[:f.slen]) + fnvHash(f.dst[:f.dlen])
-	h ^= uint64(f.typ)
-	h *= fnvPrime
+func (f Flow) FastHash() (a uint64) {
+	var b uint64
+	for i := 0; i < f.slen; i++ {
```

Comment:
This makes an assumption that `slen == dlen`, which AFAIK is 100% correct right now but may not be future-proof.  Might be safer to code around the fact that that may not be the case in the future.

Reply:
Assuming the byte array is all `0`s above the `f.slen` or `f.dlen` index, I can do the following and *just* be at the inline limit:
```
func (f Flow) FastHash() (a uint64) {
	var b uint64
	var alen int
	if f.slen > f.dlen {
		alen = f.slen
	} else {
		alen = f.dlen
	}
	for i := 0; i < alen; i++ {
...
``
```
./flows.go:160:6: can inline Flow.FastHash with cost 80 as:...
````
Will that work?